### PR TITLE
Fix typos in test_buffer.py and update links to numpy issue tracker

### PR DESCRIPTION
**Pull request by Wulian233**
suppported -> supported
and replace outdated links: http://projects.scipy.org/numpy/ticket/1910 
Because now this link is 404

### CODE REVIEW

File: Lib/test/test_buffer.py [hunk 1]
```diff
@@ -1918,7 +1917,7 @@ def test_ndarray_random(self):
                 if numpy_array:
                     shape = t[3]
                     if 0 in shape:
-                        continue # http://projects.scipy.org/numpy/ticket/1910
+                        continue
```
Comment (Eclips4) [inline Lib/test/test_buffer.py]:
```suggestion
                        continue # https://github.com/numpy/numpy/issues/2503
```

File: Lib/test/test_buffer.py [hunk 2]
```diff
@@ -1950,7 +1949,7 @@ def test_ndarray_random_invalid(self):
                     except Exception as e:
                         numpy_err = e.__class__
 
-                    if 0: # http://projects.scipy.org/numpy/ticket/1910
+                    if 0:
```
Comment (Eclips4) [inline Lib/test/test_buffer.py]:
```suggestion
                    if 0: # https://github.com/numpy/numpy/issues/2503
```

File: Lib/test/test_buffer.py [hunk 3]
```diff
@@ -1996,7 +1995,7 @@ def test_ndarray_random_slice_assign(self):
 
                 if numpy_array:
                     if 0 in lshape or 0 in rshape:
-                        continue # http://projects.scipy.org/numpy/ticket/1910
+                        continue
```
Comment (Eclips4) [inline Lib/test/test_buffer.py]:
```suggestion
                        continue # https://github.com/numpy/numpy/issues/2503
```

**Review by Eclips4 — APPROVED** [review]

**Review by aisk — APPROVED** [review]

**Comment by Wulian233** [comment]
I've completed the changes

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**